src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BDD.cs

```diff
@@ -35,24 +42,38 @@ internal class BDD : IComparable
         public readonly BDD Zero;
 
         /// <summary>
-        /// Ordinal of this bit if nonleaf else MTBDD terminal value when nonnegative
+        /// Ordinal of this bit if nonleaf else MTBDD terminal value when nonnegative.
         /// </summary>
         public readonly int Ordinal;
 
         /// <summary>
-        /// Preassigned hashcode value that respects equivalence: equivalent BDDs have equal hashcodes
+        /// Preassigned hashcode value that respects equivalence: equivalent BDDs have equal hashcodes.
         /// </summary>
-        internal readonly int hashcode;
+        private readonly int _hashcode;
 
-        internal BDD(int ordinal, BDD one, BDD zero)
+        /// <summary>
+        /// Representation of False for serialization.
+        /// </summary>
+        private static readonly long[] s_falseRepresentation = new long[] { 0 };
+
+        /// <summary>
+        /// Representation of True for serialization.
+        /// </summary>
+        private static readonly long[] s_trueRepresentation = new long[] { 1 };
+
+        public BDD(int ordinal, BDD one, BDD zero) : this(ordinal, one, zero, specialValue: false) { }
+
+        private BDD(int ordinal, BDD one, BDD zero, bool specialValue)
         {
+            if (!specialValue && ordinal < 0)
+                throw new ArgumentOutOfRangeException(nameof(ordinal), "Must be non-negative.");
             One = one;
             Zero = zero;
             Ordinal = ordinal;
-            //precompute a hashchode value that respects BDD equivalence
-            //i.e. two equivalent BDDs will always have the same hashcode
-            //that is independent of object id values of the BDD objects
-            hashcode = (ordinal, one, zero).GetHashCode();
+            // Precompute a hash code value that respects BDD equivalence i.e.
+            // two equivalent BDDs will always have the same hashcode that is
+            // independent of object id values of the BDD objects.
```

Comment:
Not worth fixing today,  but we generally don't end an inline comment with a period unless it's more than one sentence.

Reply:
Good to know, that's what Margus has been doing too I think, so I won't be "fixing" those then.